### PR TITLE
Install Intel ParallelStudio on the base_ami

### PR DESCRIPTION
**Pull request by sean-smith**
Signed-off-by: Sean Smith <seaam@amazon.com>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: resources/create_modulefile.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
+resource_name :create_modulefile
+provides :create_modulefile
+
+property :source_path, String, name_property: true
```
Comment (matteofiordarancio) [inline resources/create_modulefile.rb]: I think you should be fine just by removing this`name_property` from this parameter

File: resources/create_modulefile.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    env2 = "#{node['cfncluster']['scripts_dir']}/env2"
+    remote_file env2 do
+      source node['cfncluster']['env2']['url']
+      mode 0755
+      retries 3
+      retry_delay 5
+      not_if { ::File.exist?(env2) }
+    end
+
+    directory #{new_resource.modulefile_dir}
```
Comment (matteofiordarancio) [inline resources/create_modulefile.rb]: isn't this `directory new_resource.modulefile_dir`?

File: resources/create_modulefile.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+    # Create modulefile with env2
+    modulefile = "#{new_resource.modulefile_dir}/new_resource.modulefile"
```
Comment (matteofiordarancio) [inline resources/create_modulefile.rb]: ` "#{new_resource.modulefile_dir}/new_resource.modulefile" -> "#{new_resource.modulefile_dir}/#{new_resource.modulefile}" `

Comment (sean-smith) [reply resources/create_modulefile.rb]: changed

File: resources/create_modulefile.rb (hunk 2 above)
Comment (sean-smith) [reply resources/create_modulefile.rb]: changed

File: attributes/default.rb [hunk 4]
```diff
@@ -60,6 +60,9 @@
 default['cfncluster']['nvidia']['cuda_url'] = 'https://developer.nvidia.com/compute/cuda/10.0/Prod/local_installers/cuda_10.0.130_410.48_linux'
 # EFA
 default['cfncluster']['efa']['installer_url'] = 'https://s3-us-west-2.amazonaws.com/aws-efa-installer/aws-efa-installer-1.4.1.tar.gz'
+# ENV2
```
Comment (enrico-usai) [inline attributes/default.rb]: Could you add a more descriptive comment for "ENV2"

File: recipes/base_install.rb [hunk 5]
```diff
@@ -199,3 +199,8 @@
     end
   end
 end
+
+# Intel Runtime Libraries
+if (node['platform'] == 'centos' && node['platform_version'].to_i >= 7)
+  include_recipe "aws-parallelcluster::intel_install"
```
Comment (enrico-usai) [inline recipes/base_install.rb]: I see that _efa_install, _lustre_install etc have an underscore prefix, intel_install not.

File: recipes/intel_install.rb [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with the
+# License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+bash "install intel hpc platform" do
```
Comment (enrico-usai) [inline recipes/intel_install.rb]: We could add a brief comment on each block to describe the use of each package

File: resources/create_modulefile.rb [hunk 7]
```diff
@@ -0,0 +1,34 @@
+resource_name :create_modulefile
```
Comment (enrico-usai) [inline resources/create_modulefile.rb]: Could you add a description of what is a modulefile and for what are we using it?

File: recipes/intel_install.rb [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+  cwd node['cfncluster']['sources_dir']
+  code <<-INTEL
+    set -e
+    yum-config-manager --add-repo https://yum.repos.intel.com/intelpython/setup/intelpython.repo
+    yum -y install intelpython2 intelpython3
+  INTEL
+  creates '/opt/intel/intelpython2'
+end
+
+create_modulefile "#{node['cfncluster']['modulefile_dir']}/intelmkl" do
```
Comment (enrico-usai) [inline recipes/intel_install.rb]: Could you add a comment describing why these modulefiles are needed?

File: recipes/intel_install.rb (hunk 6 above)
Comment (sean-smith) [reply recipes/intel_install.rb]: sure! added

File: recipes/intel_install.rb (hunk 8 above)
Comment (sean-smith) [reply recipes/intel_install.rb]: sure! (they're not really needed 🤫) however it greatly improves the visibility of each of the packages to have them. without the modulefiles, it's really unclear what's installed with psxe

File: resources/create_modulefile.rb (hunk 7 above)
Comment (sean-smith) [reply resources/create_modulefile.rb]: Added!

File: recipes/base_install.rb (hunk 5 above)
Comment (enrico-usai) [reply recipes/base_install.rb]: I agree, thank you for the explanation.

**Comment by enrico-usai** [comment]
The patch is approved for me, but let's wait for integration test before merging.

**Pull request merged**